contrib_bots/lib/gmail.py

```diff
@@ -0,0 +1,174 @@
+    '''
+
+    def usage(self):
+        return '''
+            This plugin will allow users to send emails from a
+            personal gmail account. Users should preface
+            messages with "@gmail".
+
+            Enter messages in the following format:
+            "@gmail --recipient <recipient_email> --subject <subject> --label <label> --body
+            <body>". The --subject and --label tags are optional and can be left blank.
+
+            Before running this, make sure to create a bot and store its information
+            in ~/.zuliprc.
+            '''
+
+    def triage_message(self, message, client):
+        # return True if we want to (possibly) response to this message
+
+        original_content = message['content']
+        self.bot_email = client.email
+
+        # This next line of code is defensive
+        if message['display_recipient'] == 'gmail':
+            return False
+        is_follow_up = original_content.startswith('@gmail')
+        return is_follow_up
+
+    def handle_message(self, message, client, state_handler):
+        original_content = message['content']
+        original_sender = message['sender_email']
+
+        values = parse_input(original_content)
+        body = values['body']
+        label = values['label']
+        subject = values['subject']
+        recipient = values['recipient']
+
+        service = get_service()
+        email = create_message(body, subject, recipient)
+        result = send_message(service, email)
+        print(result)
+
+        label_result = -1
+        if (label != ''):
+            create_label(service, label)
+            label_id = get_label_id(service, label)
+            label_result = add_label(service, label_id, result['id'])
+
+        if ('SENT' in result['labelIds']):
+            if (label_result == 0):
+                client.send_message(dict(
+                    type = 'private',
+                    to = original_sender,
+                    sender = self.bot_email,
+                    content = 'Email sent to {}'.format(recipient)
+                ))
+            else:  # Enters this block if the label couldn't be added or there wasn't a label
+                client.send_message(dict(
+                    type = 'private',
+                    to = original_sender,
+                    sender = self.bot_email,
+                    content = 'Email sent to {} without label'.format(recipient)
+                ))
+        else:
+            client.send_message(dict(
+                type = 'private',
+                to = original_sender,
+                sender = self.bot_email,
+                content = 'Could not send email to {}. Please try again.'.format(recipient)
+            ))
+
+
+def parse_input(user_input):
+    # Input is of the following format:
+    # @gmail --recipient <recipient_email> --subject <subject> --label <label> --body <body>
+    body = user_input.split('--body')[1].strip()
```

Comment:
@fazerlicourice7 You can use a tool like [PyRegex](http://www.pyregex.com/) to test out different approaches. As for learning the regular expressions themselves, I think Python docs have a pretty good explanation. 

Should you have any specific problems, chat with us on Zulip.